### PR TITLE
Fix a failed `BlockAllOperations` leaving update realm in unretrieved state

### PR DESCRIPTION
**Pull request by peppy**
If the operation timed out on..

```csharp
throw new TimeoutException(@"Took too long to acquire lock");
```

..from an update thread, it would not restore the update context.

The next call would then fail on the assert that ensures a non-null context
in such cases.

Can add test coverage if required.

- [x] Depends on #16593 for conflict avoidance.

### CODE REVIEW

File: osu.Game/Database/RealmAccess.cs [hunk 1]
```diff
@@ -595,8 +595,8 @@ public IDisposable BlockAllOperations()
                 {
                     if (updateRealm == null)
                     {
-                        // null realm means the update thread has not yet retrieved its instance.
-                        // we don't need to worry about reviving the update instance in this case, so don't bother with the SynchronizationContext.
+                        // null context means the update thread has not yet retrieved its context.
+                        // we don't need to worry about reviving the update context in this case, so don't bother with the SynchronizationContext.
```
Comment (bdach) [inline osu.Game/Database/RealmAccess.cs:599]: guessing that this here is an inadvertent partial revert of f30894840c709505464cc2b903f4a8870feb7fd4

**Review by bdach — COMMENTED** [review]
Solution works in my head but I have not tested manually and test coverage would definitely be a bonus.

File: osu.Game/Database/RealmAccess.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Database/RealmAccess.cs]: Fixed this in the merge commit.

**Comment by peppy** [comment]
Added test coverage, really easy one.

**Comment by smoogipoo** [comment]
Test failures on this one

**Comment by peppy** [comment]
Not sure how I saw it passing. Won't work because the `SynchronizationContext` is never run.

**Review by smoogipoo — APPROVED** [review]
Looks fine in theory.

**Pull request merged**